api/src/pcapi/core/offers/api.py

```diff
@@ -321,6 +324,17 @@ def update_offer(
     if not modifications:
         return offer
 
+    if (UNCHANGED, UNCHANGED) != (withdrawalType, withdrawalDelay):
+        try:
+            changed_withdrawalType = withdrawalType if withdrawalType != UNCHANGED else None
+            changed_withdrawalDelay = withdrawalDelay if withdrawalDelay != UNCHANGED else None
```

Comment:
Si la valeur n'a pas changé, il faudrait prendre comme "base" la valeur **actuelle** (et pas `None`), non ?

```suggestion
            changed_withdrawalType = withdrawalType if withdrawalType != UNCHANGED else offer.
            changed_withdrawalDelay = withdrawalDelay if withdrawalDelay != UNCHANGED else offer.withdrawalDelay
```

Reply:
merci !

j'avais oublié / mal fait ce cas

j'ai ajouté un test pour m'en assurer